src/message/moveMessage.js

```diff
@@ -0,0 +1,225 @@
+import { Input } from '../common';
+import type { AppNavigationProp } from '../nav/AppNavigator';
+import { getAuth, getStreams } from '../selectors';
+import { useSelector } from '../react-redux';
+import { showErrorAlert } from '../utils/info';
+import { IconBack } from '../common/Icons';
+import type { Narrow, Message } from '../types';
+import TopicAutocomplete from '../autocomplete/TopicAutocomplete';
+import { streamNameOfNarrow } from '../utils/narrow';
+
+type Props = $ReadOnly<{|
+  navigation: AppNavigationProp<'move-message'>,
+  route: RouteProp<'move-message', {| message: Message, messageNarrow: Narrow, isadmin: boolean |}>,
+|}>;
+
+const inputMarginPadding = {
+  paddingHorizontal: 8,
+  paddingVertical: Platform.select({
+    ios: 8,
+    android: 2,
+  }),
+};
+
+export default function MoveMessage(props: Props): Node {
+  const themeContext = useContext(ThemeContext);
+  const backgroundColor = themeContext.backgroundColor;
+  const cardColor = themeContext.cardColor;
+  const auth = useSelector(getAuth);
+  const streams = useSelector(getStreams);
+  const isadmin = props.route.params.isadmin;
+  const id = props.route.params.message.id;
+  const names = [];
+  const [subject, setSubject] = useState(props.route.params.message.subject);
+  const [propagate_mode, setPropagateMode] = useState('javachange_one');
+  const [stream, setStream] = useState(streamNameOfNarrow(props.route.params.messageNarrow));
+  const [topicFocus, setTopicFocus] = useState(false);
+  const messageInputRef = React.createRef<$FlowFixMe>();
+  const topicInputRef = React.createRef<$FlowFixMe>();
+
+  streams.map(item => names.push({ value: item.name }));
+
+  const styles = {
+    parent: {
+      backgroundColor: cardColor,
+    },
+    layout: {
+      margin: 10,
+    },
+    title: {
+      fontSize: 18,
+      color: backgroundColor === 'white' ? 'black' : 'white',
+    },
+    autocompleteWrapper: {
+      position: 'absolute',
+      top: 0,
+      width: '100%',
+    },
+    composeText: {
+      flex: 1,
+      paddingVertical: 8,
+    },
+    topicInput: {
+      height: 50,
+      backgroundColor,
+      ...inputMarginPadding,
+    },
+    viewTitle: {
+      display: 'flex',
+      flexDirection: 'row',
+      justifyContent: 'space-between',
+      alignItems: 'center',
+      height: 50,
+      paddingHorizontal: 10,
+      marginBottom: 20,
+    },
+    textColor: {
+      color: backgroundColor === 'white' ? 'black' : 'white',
+    },
+    picker: { backgroundColor, marginBottom: 20 },
+    submitButton: {
+      marginTop: 10,
+      paddingTop: 15,
+      paddingBottom: 15,
+      marginLeft: 30,
+      marginRight: 30,
+      backgroundColor: BRAND_COLOR,
+      borderRadius: 10,
+      borderWidth: 1,
+    },
+  };
+
+  const updateTextInput = (textInput, text) => {
+    if (textInput === null) {
+      // Depending on the lifecycle events this function is called from,
+      // this might not be set yet.
+      return;
```

Comment:
This comment appears to be copy-pasted from `ComposeBox.js`. Is the comment true here? What does it mean?

… Reading a bit more, it looks like in fact `textInput` will *always* be null here -- it comes from `topicInputRef.current`, and we never set `topicInputRef` or pass it somewhere that could end up setting it. So this function never does anything.